anomalib/utils/callbacks/post_processing_configuration.py

```diff
@@ -0,0 +1,63 @@
+"""Post-Processing Configuration Callback."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+from typing import Optional
+
+import torch
+from pytorch_lightning import Callback, LightningModule, Trainer
+from pytorch_lightning.utilities.cli import CALLBACK_REGISTRY
+
+from anomalib.models.components.base.anomaly_module import AnomalyModule
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["PostProcessingConfigurationCallback"]
+
+
+@CALLBACK_REGISTRY
+class PostProcessingConfigurationCallback(Callback):
+    """Post-Processing Configuration Callback.
+
+    Args:
+        normalization_method(Optional[str]): Normalization method. <None, min_max, cdf>
+        adaptive_threshold (bool): Flag indicating whether threshold should be adaptive.
+        default_image_threshold (Optional[float]): Default image threshold value.
+        default_pixel_threshold (Optional[float]): Default pixel threshold value.
+    """
+
+    def __init__(
+        self,
+        normalization_method: str = "min_max",
+        adaptive_threshold: bool = True,
+        default_image_threshold: Optional[float] = None,
+        default_pixel_threshold: Optional[float] = None,
+    ) -> None:
+        super().__init__()
+        self.normalization_method = normalization_method
+
+        assert (
+            adaptive_threshold or default_image_threshold is not None and default_pixel_threshold is not None
+        ), "Default thresholds must be specified when adaptive threshold is disabled."
+
+        self.adaptive_threshold = adaptive_threshold
```

Comment:
We could consider replacing this parameter with a selectable called `thresholding_method` with the options `fixed` and `adaptive`. This might be a bit less confusing (for the unknowing reader, `adaptive_threshold` could sound like it holds a threshold value). Later we could extend it with `synthetic` option.

Reply:
I was also having a similar idea in my mind. I was not quite sure how exactly to address fixed image and pixel thresholds though

Reply:
It would be the equivalent of `adaptive_threshold: false` in the current implementation

Reply:
yeah, but let's say we set `thresholding_method` to `fixed`. We would still need to set `fixed_image_threshold` and `fixed_pixel_threshold`, right?

Reply:
yeah, but I don't see any problems with that. If users want to use a fixed threshold then they will have to provide a value for that. If they don't know which value to provide then they should just use the adaptive threshold.